conda_forge_metadata/artifact_info/info_json.py

```diff
@@ -138,8 +138,19 @@ def info_json_from_tar_generator(
             data["conda_build_config"] = YAML.load(
                 _extract_read(tar, member, default="{}")
             )
-        elif path.name == "files":
-            files = _extract_read(tar, member, default="").splitlines()
+        elif path.name == "variant_config.yaml":
+            data["conda_build_config"] = YAML.load(
+                _extract_read(tar, member, default="{}")
+            )
+        elif path.name == "paths.json":
+            paths = json.loads(_extract_read(tar, member, default="{}"))
+            paths_version = paths.get("paths_version", 1)
+            if paths_version != 1:
+                warnings.warn(
+                    f"Unrecognized paths_version {paths_version} in paths.json",
+                    RuntimeWarning,
+                )
+            files = [p.get("_path", "") for p in paths.get("paths", [])]
             if skip_files_suffixes:
```

Comment:
The if-clause can be removed because `endswith` behaves correctly on empty tuples.

I verified via a GitHub search that no one passes weird stuff like `None`, violating the type annotation.

Reply:
I don't see the harm in keeping this clause just in case something weird gets passed in.

Reply:
No strong opinion here, but, generally, I think we actually *want* client code to break if it doesn't adhere to the API defined by type annotations. In my experience, this is the only way to reduce reliance on undocumented code behaviour, which is technical debt we don't want to have.